Add empty Apptainer definition files

milljm opened the pull request:
Add empty Apptainer recipe/definition files so that module_hash functions correctly.

Closes #22505

Commit:
Add empty Apptainer definition files

Add empty Apptainer recipe/definition files so that module_hash
functions correctly.

Closes #22505

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1214937/) on 0952cdc wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/22506/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1214950/) on 0952cdc wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>a55c19</th>
      <th colspan=4>#22506 0952cd</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>84.22%</td>
      <td>84.23%</td>
      <td>+0.01%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>80238</td>
      <td>80243</td>
      <td>+5</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15032</td>
      <td>15027</td>
      <td>-5</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/22506/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/22506/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/22506/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/22506/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/22506/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/22506/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/22506/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/22506/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/22506/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/22506/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/22506/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/22506/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/22506/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/22506/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/22506/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/22506/coverage/navier_stokes)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/22506/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/22506/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/22506/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/22506/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/22506/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/22506/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/22506/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/22506/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/22506/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/22506/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/22506/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/22506/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/22506/coverage/xfem)

</details>



This comment will be updated on new commits.

Commit:
Rename MPI to MPICH, add ARCH

Apply ARCH identification.

We're using (going to use) MPICH in Apptainer. Conda is MPICH. Spack
is MPICH. Might as well just call the whole thing MPICH.

Naming it 'mpich' also helps with Conda package dependency tracking.
As that tool checks for moose/conda/<ZIP_KEY NAME> presence.

Commit:
Always forgetting the Unittests

Commit:
Simplify the table building process

Support printing tags with dependencies

Commit:
Need to reverse the list

Reverse the list when dealing with --dependencies

Commit:
Add missing tag only feature

Commit:
Add json/yaml output support

Commit:
Add oras format specific output

Default to printing HASH only. Add feature to print oras friendly URI.

Refactor meta dictionary a bit.

Commit:
Supply 'libmesh' as the default when appropriate

I don't think we will ever supply a built version/container of only
moose which others will 'depend on' in order to build their application.

When this scenario occurs, its best to instead supply libmesh as their
dependency.

loganharbour commented:
Completed in https://github.com/idaholab/moose/pull/22778
